src/containers/SecurityDetails.js

```diff
@@ -0,0 +1,293 @@
+  if (fieldProperties.notEditable.indexOf(attr)<0){
+    return 0;
+  }
+  else if (this.isPrimitive(attr)){
+    return 1;
+  }
+  else if (this.isPrimitive(attr) && !this.canWrite(attr)){
+    return 1;
+  }
+  else if (this.isPrimitive(attr) && this.canWrite(attr)){
+    return 2;
+  }
+  else if (!this.isPrimitive(attr) && !this.canWrite(attr)){
+    return 3;
+  }
+  else if (!this.isPrimitive(attr) && this.canWrite(attr)){
+    return 4;
+  }
+  else {
+    return 5;
+  }
+}
+
+class SecurityDetails extends Component {
+
+  getAttributeNames = (entity) => {
+    var attributeList = [];
+    for (var attribute in entity) {
+      if (entity.hasOwnProperty(attribute)) {
+        attributeList.push(attribute);
+        if (entity[attribute] instanceof Object) {
+          var sublist = this.getAttributeNames(entity[attribute]);
+          for (var i in sublist) {
+            attributeList.push(attribute + '.' + sublist[i]);
+          }
+        }
+      }
+    }
+    return attributeList;
+  }
+
+  canWrite(attribute) {
+    attribute = attribute ? attribute : 'actions.self';
+    return (this.props.entityActionsPolicies.policies
+      && this.props.entityActionsPolicies.policies[attribute]
+      && this.props.entityActionsPolicies.policies[attribute].write)
+
+  }
+
+  canRead(attribute) {
+    attribute = attribute ? attribute : 'actions.self';
+    return (this.props.entityActionsPolicies.policies
+      && this.props.entityActionsPolicies.policies[attribute]
+      && this.props.entityActionsPolicies.policies[attribute].read)
+  }
+
+  getDeleteButton(params) {
+    return (
+      <FloatingActionButton mini={true}  id={'delete_' + params.id + '_' + params.attribute}
+                    key={params.id + '_' + params.attribute}
+                    style={deleteButtonStyle}
+                    label='Delete'
+                    onClick={() => {
+                      this.props.deleteAttribute(params)
+                    }}>
+            <ContentRemove />
+      </FloatingActionButton>
+    )
+  }
+
+  getAddAttributeFields(parent) {
+    if (this.canWrite(parent)) {
+      var newAttribute = {};
+      var attributeName = parent ? parent + 'newAttr' : 'newAttr';
+      var attributeValue = parent ? parent + 'newAttrVal' : 'newAttrVal';
+      return (
+        <div>
+          <input onFocus={(event) => {
+            event.stopPropagation()
+          }} onClick={(event) => {
+            event.stopPropagation()
+          }}
+                 id={attributeName} placeholder='New Attribute'/>
+          <input onFocus={(event) => {
+            event.stopPropagation()
+          }} onClick={(event) => {
+            event.stopPropagation()
+          }}
+                 id={attributeValue} placeholder='Value'/>
+           <FloatingActionButton mini={true}  style={addButtonStyle}
+                         label='Add'
+                         onClick={(event) => {
+                           event.stopPropagation();
```

Comment:
See my comment below about dom methods, you should use internal state to keep track of things.